### PR TITLE
Pause running workflows when stopping beeflow with 'beeflow core stop'

### PR DESCRIPTION
**Pull request by pagrubel**
This PR pauses any running or waiting workflows when stopping beeflow.
Tasks/jobs that are running or pending when beeflow is stopped will be updated, if possible, once beeflow starts up again.
Addresses #783

### CODE REVIEW

**Comment by pagrubel** [comment]
I need to do some overnight testing on this branch.

**Comment by pagrubel** [comment]
To test this for CLI (on darwin) you may want two screens both in the poetry env for this branch:
  - Make sure use_commands in the slurm portion of bee.conf is True.
  - Start beeflow: beeflow core start
  - submit a workflow, clamr example works well, the checkpoint workflow is also good
      ``watch -n 2 query <wf_id>`` to make sure the task is pending or running. 
       It is helpful to do this in a separate screen and keep it up even when you stop beeflow, or when you restart it
  - verify that the clamr step is running and get the job id via ``squeue -u <username> `` or the task manager log
  - issue ``beeflow core stop`` while clamr step is running
  - ``watch -n 5 show job <job_id>`` Wait until this gives a no job type error
  - Then start beeflow up again ``beeflow core start``. The query screen will show that clamr has completed.
  - The workflow has been paused so to finish it just submit the  ``beeflow resume <wf_id>``

To test for slurmrestd:
- Make sure use_commands in the slurm portion of bee.conf is False.
  - Start beeflow: beeflow core start
  - submit a workflow, clamr example works well, the checkpoint workflow is also good
      ``watch -n 2 query <wf_id>`` to make sure the task is pending or running. 
       It is helpful to do this in a separate screen and keep it up even when you stop beeflow, or when you restart it
  - verify that the clamr step is running and get the job id via ``squeue -u <username> `` or the task manager log
  - issue ``beeflow core stop`` while clamr step is running
  - ``watch -n 5 squeue -u <userid>`` Wait until the clamr job is off the screen
  - Then start beeflow up again ``beeflow core start``. The query screen will show that clamr has completed.
  - The workflow has been paused so to finish it just submit the  ``beeflow resume <wf_id>``

**Comment by pagrubel** [comment]
Adding this thought here to implement later: The output from "sacct -j <job_id>" is in column form where the job_id is a row heading and Job State is a column heading. That should be used instead of the location in case another column or row is added in the future.

**Comment by pagrubel** [comment]
@jtronge What happens if sacct throws an exception with job_state? I was hoping to get it out of the queue but not sure it will be.

**Comment by pagrubel** [comment]
@jtronge I tested this on the prod system and it works. The only additional feature we may want to add at some point is for the user to be able to resume all paused workflows with one command.

File: beeflow/task_manager/background.py [hunk 1]
```diff
@@ -94,15 +94,14 @@ def update_jobs(db):
...
-                log.info(f'Task {task.name} in state {new_job_state}')
                 log.info(f'Resubmitting task {task.name}')
                 db.job_queue.remove_by_id(id_)
                 job_state = submit_task(db, worker, task)
                 db.update_queue.push(task.workflow_id, task.id, job_state)
             else:
                 db.update_queue.push(task.workflow_id, task.id, new_job_state)
 
-        if job_state in ('ZOMBIE', 'COMPLETED', 'CANCELLED', 'FAILED', 'TIMEOUT', 'TIMELIMIT'):
+        if job_state in ('UNKNOWN', 'COMPLETED', 'CANCELLED', 'FAILED', 'TIMEOUT', 'TIMELIMIT'):
```
Comment (jtronge) [inline beeflow/task_manager/background.py:104]: Does this get rid of `ZOMBIE` altogether?

**Review by jtronge — COMMENTED** [review]
I think this looks good. Since I made a couple of commits, should someone else approve as well?

**Comment by jtronge** [comment]
> @jtronge What happens if sacct throws an exception with job_state? I was hoping to get it out of the queue but not sure it will be.

Oh sorry, just saw your question. I think this depends on what the calling code is doing. The TM background code could try to catch `WorkerError` and remove the job in that case.

File: beeflow/task_manager/background.py (hunk 1 above)
Comment (pagrubel) [reply beeflow/task_manager/background.py:104]: I think ZOMBIE was meant for Workflow states

Comment (pagrubel) [reply beeflow/task_manager/background.py:104]: Not for jobs

**Comment by pagrubel** [comment]
> > @jtronge What happens if sacct throws an exception with job_state? I was hoping to get it out of the queue but not sure it will be.
> 
> Oh sorry, just saw your question. I think this depends on what the calling code is doing. The TM background code could try to catch `WorkerError` and remove the job in that case.

Should we put WIP back on do this. I'm not sure there is any way to test it.

**Review by aquan9 — APPROVED** [review]
This all looks ok to me.

**Pull request merged**